### PR TITLE
feat: Bitwarden Secrets provider

### PR DESCRIPTION
**Pull request by carnei-ro**
adds Bitwarden Secrets  
closes https://github.com/helmfile/vals/issues/427

This provider uses the [Bitwarden SDK Go](https://github.com/bitwarden/sdk/tree/main/languages/go#bitwarden-sdk-in-go), which needs GCO enabled and musl-tools to compile (linux), check [this doc](https://github.com/bitwarden/sdk/blob/main/languages/go/INSTRUCTIONS.md).

Use `ref+bws://<PROJECT_NAME>/<SECRET_NAME>` to fetch the secret value.

```bash
$ make build
CGO_ENABLED=1 CC=clang CXX=clang++ go build -ldflags '-X main.version=v0.37.3-56-g8a1c52c-dirty -X main.commit=8a1c52c880fe3a40e0c65035b6ab5f8e742b7534' -o bin/vals ./cmd/vals

$ cat myyaml.yaml 
baz:
  mykey:
    simple_secret: ref+bws://vals-test/fooKey
    yaml: |
      ref+bws://vals-test/myYAML
    parsing_yaml_secret: ref+bws://vals-test/myYAML#baz

$ cat myyaml.yaml | bin/vals eval
baz:
  mykey:
    parsing_yaml_secret: qux
    simple_secret: myValue
    yaml: |
      foo: bar
      baz: qux

$ env | grep BWS_
BWS_ORGANIZATION_ID=d2f88df6-REDA-CTED-0000-000000000000
BWS_ACCESS_TOKEN=0.b5a15cb4-REDA-CTED-0000-000000000000.abcdefghijklmnopqrstuvwxyzABCD:aSBsb3ZlIHZhbHMgISAhCg==
```

The test is passing:

```bash
$ go test -run TestValues_BitwardenSecrets_String .
ok      github.com/helmfile/vals        7.151s
```

image

### CODE REVIEW

File: pkg/providers/bitwardensecrets/bitwardensecrets.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	ProjectName      string
+	ProjectID        string
+}
+
+func New(l *log.Logger, cfg api.StaticConfig) *provider {
+	p := &provider{
+		log: l,
+	}
+
+	p.ApiUrl = cfg.String("api_url")
```
Comment (zhaque44) [inline pkg/providers/bitwardensecrets/bitwardensecrets.go]: extract this repeated logic into a helper func to simplify

File: pkg/providers/bitwardensecrets/bitwardensecrets.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	bitwardenClient, _ := sdk.NewBitwardenClient(&p.ApiUrl, &p.IdentityURL)
+
+	err = bitwardenClient.AccessTokenLogin(p.AccessToken, nil)
+	if err != nil {
+		return "", err
+	}
+
+	p.OrganizationUUID, err = uuid.FromString(p.OrganizationID)
+	if err != nil {
+		panic(err)
```
Comment (zhaque44) [inline pkg/providers/bitwardensecrets/bitwardensecrets.go]: replace the panic calls with proper error handling, returns the err instead

File: pkg/providers/bitwardensecrets/bitwardensecrets.go (hunk 1 above)
Comment (carnei-ro) [reply pkg/providers/bitwardensecrets/bitwardensecrets.go]: do you have an example on any other provider?

File: vals_bitwardensecrets_test.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	}
+
+	for i := range testcases {
+		tc := testcases[i]
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			config := config2.Map(tc.config)
+
+			vals, err := Load(config)
+			if err != nil {
+				t.Fatalf("%v", err)
```
Comment (zhaque44) [inline vals_bitwardensecrets_test.go]: Add more context to the error message by including the test case index, ex: **t.Fatalf("failed to load config for testcase %d: %v", i, err)**

File: pkg/providers/bitwardensecrets/bitwardensecrets.go (hunk 1 above)
Comment (zhaque44) [reply pkg/providers/bitwardensecrets/bitwardensecrets.go]: I pushed an example, of what it should be like

Comment (carnei-ro) [reply pkg/providers/bitwardensecrets/bitwardensecrets.go]: ok, I've changed.

**Comment by yxxhero** [comment]
@zhaque44 this PR broken many build process.

**Pull request merged**